### PR TITLE
[nit][DCP][DSD] Remove Unused Variables in test_state_dict.py

### PR DESCRIPTION
**Pull request by wz337**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121204

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @tianyu-l @wconstab @yf225 @chauhang @LucasLLC

### CODE REVIEW

**Review by fegin — COMMENTED** [review]
Why does this change cause FSDP compilation state_dict error? Is the error pre-existed?

**Comment by wz337** [comment]
> Why does this change cause FSDP compilation state_dict error? Is the error pre-existed?

Seems like so. I am having this error locally and I don't think it's related to the change, so I push just push these two-line changes definitely unrelated to double check.

File: test/distributed/checkpoint/test_state_dict.py [hunk 1]
```diff
@@ -106,9 +106,6 @@ def _test_save_load(
         # Then finally we can call set_state_dict().
         if not isinstance(dist_optim, list):
             dist_optim = [dist_optim]
-        curr_dist_msd, curr_dist_osd = get_state_dict(
```
Comment (Skylion007) [inline test/distributed/checkpoint/test_state_dict.py:109]: Hmm, should have been caught by F841

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**